src/main/java/com/github/nylle/javafixture/specimen/TimeSpecimen.java

```diff
@@ -0,0 +1,72 @@
+package com.github.nylle.javafixture.specimen;
+
+import java.time.Duration;
+import java.time.MonthDay;
+import java.time.Period;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.temporal.TemporalAmount;
+import java.util.Random;
+
+import com.github.nylle.javafixture.ISpecimen;
+import com.github.nylle.javafixture.SpecimenException;
+
+public class TimeSpecimen<T> implements ISpecimen<T> {
+
+    private final Class<T> type;
+    private final Random random;
+
+    public TimeSpecimen( Class<T> type ) {
+        if ( type == null ) {
+            throw new IllegalArgumentException( "type: null" );
+        }
+
+        if ( !isTimeClass( type ) ) {
+            throw new IllegalArgumentException( "type: " + type.getName() );
+        }
+
+        this.type = type;
+        this.random = new Random();
+    }
+
+    private static boolean isTimeClass( Class type ) {
```

Comment:
If `isTimeSpecimen` has been moved to `ReflectionHelper`, could it be re-used here instead?

Reply:
yes. Thanks for spotting the duplicate code